### PR TITLE
#2059 Also consider indirectly dependent objects when determining when a case was last updated

### PR DESCRIPTION
**Pull request by sergiupacurariu**

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseFacade.java [hunk 1]
```diff
@@ -42,6 +42,8 @@
 
 	List<CaseDataDto> getAllActiveCasesAfter(Date date);
 
+	List<CaseDataDto> getAllActiveCasesAfter(Date date, Boolean includeAdditionalParams);
```
Comment (MateStrysewskeSym) [inline sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseFacade.java:45]: Rename new parameter to "includeExtendedChangeDateFilters" and add a comment that explains what exactly these additional filters are (do the renaming for all other methods where this parameter occurs as well)

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java [hunk 2]
```diff
@@ -714,6 +711,19 @@ public Predicate createChangeDateFilter(CriteriaBuilder cb, From<?, Case> casePa
...
+			Join<Case, Sample> caseSampleJoin = casePath.join(Case.SAMPLES, JoinType.LEFT);
+			dateFilter = cb.or(dateFilter,
+					greaterThanAndNotNull(cb, caseSampleJoin.get(AbstractDomainObject.CHANGE_DATE), date));
+
+			Join<Sample, PathogenTest> samplePathogenTestJoin = caseSampleJoin.join(Sample.PATHOGENTESTS, JoinType.LEFT);
+			dateFilter = cb.or(dateFilter,
+					greaterThanAndNotNull(cb, samplePathogenTestJoin.get(AbstractDomainObject.CHANGE_DATE), date));
+
+			Join<Case, Person> casePersonJoin = casePath.join(Case.PERSON, JoinType.LEFT);
+			dateFilter = cb.or(dateFilter, greaterThanAndNotNull(cb, casePersonJoin.get(AbstractDomainObject.CHANGE_DATE), date));
```
Comment (MateStrysewskeSym) [inline sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java:724]: You also need to check the person address change date

File: sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java [hunk 3]
```diff
@@ -45,6 +45,12 @@
 		return FacadeProvider.getCaseFacade().getAllActiveCasesAfter(new Date(since));
 	}
 
+	@GET
+	@Path("/allIncludeAdditionalParams/{since}")
+	public List<CaseDataDto> getAllCasesInCludeAdditionalObjects(@PathParam("since") long since) {
```
Comment (MateStrysewskeSym) [inline sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java:50]: Rename to getAllCasesWithExtendedChangeDateFilters

File: sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java [hunk 4]
```diff
@@ -45,6 +45,12 @@
 		return FacadeProvider.getCaseFacade().getAllActiveCasesAfter(new Date(since));
 	}
 
+	@GET
+	@Path("/allIncludeAdditionalParams/{since}")
```
Comment (MateStrysewskeSym) [inline sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java:49]: Rename to allWithExtendedChangeDateFilters

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseFacade.java (hunk 1 above)
Comment (sergiupacurariu) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/CaseFacade.java:45]: done

File: sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java (hunk 3 above)
Comment (sergiupacurariu) [reply sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java:50]: changed

File: sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java (hunk 4 above)
Comment (sergiupacurariu) [reply sormas-rest/src/main/java/de/symeda/sormas/rest/CaseResource.java:49]: changed

File: sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java (hunk 2 above)
Comment (sergiupacurariu) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java:724]: updated

**Pull request merged**